src/frame/proxy.rs

```diff
@@ -0,0 +1,27 @@
+use codec::{Decode, Encode};
+use core::marker::PhantomData;
+use substrate_subxt::{
+    system::{System, SystemEventsDecoder},
+    Encoded,
+};
+use substrate_subxt_proc_macro::{module, Call};
+
+/// The subset of the `frame_proxy::Trait` that a client must implement.
+#[module]
+pub trait Proxy: System {
+    type ProxyType: 'static + Encode + Decode + Send + Default + Sync;
+}
+
+/// Dispatch the given `call` from an account that the sender is authorised for through
+/// `add_proxy`
+#[derive(Clone, Debug, Eq, PartialEq, Call, Encode)]
+pub struct ProxyCall<'a, T: Proxy> {
+    /// The account that the proxy will make a call on behalf of.
+    pub real: <T as System>::AccountId,
+    /// Specify the exact proxy type to be used and checked for this call.
+    pub force_proxy_type: Option<T::ProxyType>,
+    /// The call to be made by the `real` account.
+    pub call: &'a Encoded,
+    /// Runtime marker.
+    pub _runtime: PhantomData<T>,
```

Comment:
```pub _runtime: PhantomData<T>,``` should be in the first place?

Reply:
yes, it must be in the first place

Reply:
If the target struct of `Store` or `Call` has fields contains the generic `T`, we don't have to add the `_runtime` field to the struct, because the `PhatomData<T>` only marks the relationship between `Store`/`Call` and the frame trait.